docs/MANUAL.md

```diff
@@ -0,0 +1,215 @@
+
+An improperly configured `sonar-project.properties` is the number one cause of analysis issues and
+inconsistencies when using SonarDelphi - if you get it right, there's not much SonarDelphi can't handle.
+
+### Advanced project configuration
+
+SonarQube exposes configuration for code duplication thresholds, which can be useful to adjust for
+verbose languages like Delphi.
+
+```ini
+# By default, a piece of code is considered duplicated as soon as there are at least
+# 100 successive and duplicated tokens spread over at least 10 lines of code.
+sonar.cpd.delphi.minimumTokens=200
+sonar.cpd.delphi.minimumLines=20
+```
+
+For more details on SonarQube configuration options, see
+[the SonarQube documentation](https://docs.sonarsource.com/sonarcloud/advanced-setup/analysis-parameters/).
+
+In addition to `sonar.tests`, SonarDelphi allows you to configure certain types and their methods as test code.
+This is designed for compatibility with common Delphi test frameworks and can be configured in two ways:
+
+```ini
+# Code within this type and all descendants is considered test code.
+# Defaults to DUnit's test base type: TestFramework.TTestCase.
+sonar.delphi.testType=TestFramework.TTestCase
+# Code within any type annotated with this attribute is considered test code.
+# Defaults to DUnitX's test attribute: DUnitX.Attributes.TestFixtureAttribute.
+sonar.delphi.testAttribute=DUnitX.Attributes.TestFixtureAttribute
+```
+
+For the full list of SonarDelphi configuration options, see [here](CONFIGURATION.md).
+
+## Run a SonarDelphi scan
+
+Once your project is configured, you simply
+[run the scanner](https://docs.sonarsource.com/sonarqube/latest/analyzing-source-code/scanners/sonarscanner/)
+in the same directory as your `sonar-project.properties`.
+
+After an analysis, the project page on SonarQube will refresh with updated issues and metrics.
+Browsing the issues, you have the ability to update them if needed - assign them to other users, add comments,
+or change their status (e.g. accepting issues you won't fix or hiding false positives).
+
+If there are a large number of issues for rules that don't suit your codebase - for example, issues complaining
+your classes don't begin with `T` when you have a different convention - you may benefit from
+[customizing SonarDelphi's analysis rules](#configure-analysis-rules).
+
+## Configure analysis rules
+
+The rules applied during a scan are dictated by the project's
+[quality profile](https://docs.sonarsource.com/sonarqube/latest/instance-administration/quality-profiles/).
+
+For Delphi, the default quality profile ("Sonar way") contains sensible default rules that are generally
+accepted to be best practice. These may not be helpful to apply to your codebase if it has its own conventions that
+conflict with the rules. Some common quality profile changes include:
+
+* Setting `excludeApi` to `true` (allowing "unused" public API) on `UnusedConstant`, `UnusedField`,
```

Comment:
I think we can lose some information here in the name of brevity.
```suggestion
* Setting `excludeApi` to `true` on `UnusedConstant`, `UnusedField`,
```

Reply:
I don't think brevity's the name of the game here, to be honest.

This manual is (ideally) written so that a new user can read through and be able to configure their analysis exactly how they want. `excludeApi` is very important to configure correctly if you're scanning a library, and so we should make clear what it does and when you would want it.

Odds are that the user is trying to configure as little as possible to get what they want, so if we just say "setting the nebulous property `excludeApi` to `true` is something that lots of people do!" then odds are that a user won't.

Reply:
Sure, that's fine with me.